beacon_node/beacon_chain/src/block_verification.rs

```diff
@@ -328,6 +330,48 @@ pub enum BlockError<E: EthSpec> {
     InternalError(String),
 }
 
+impl<E: EthSpec> BlockError<E> {
+    pub fn rpc_scoring(&self) -> ErrorCategory {
+        match self {
+            Self::ParentUnknown { .. } => ErrorCategory::internal_recoverable(),
+            Self::FutureSlot { .. } => ErrorCategory::internal_recoverable(),
+            Self::StateRootMismatch { .. } => ErrorCategory::malicious_non_recoverable(),
+            Self::GenesisBlock => ErrorCategory::internal_non_recoverable(),
+            Self::WouldRevertFinalizedSlot { .. } => ErrorCategory::internal_non_recoverable(),
+            Self::NotFinalizedDescendant { .. } => ErrorCategory::internal_non_recoverable(),
+            Self::BlockIsAlreadyKnown { .. } => ErrorCategory::internal_non_recoverable(),
+            Self::BlockSlotLimitReached => ErrorCategory::malicious_non_recoverable(),
+            Self::IncorrectBlockProposer { .. } => ErrorCategory::malicious_non_recoverable(),
+            Self::ProposalSignatureInvalid => ErrorCategory::malicious_recoverable(),
+            Self::UnknownValidator { .. } => ErrorCategory::malicious_non_recoverable(),
+            // TODO: Labeling as recoverable as it may be the proposer signature. We should check
+            // Which one is it and label as non recoverable if the proposer signature is correct.
```

Comment:
if the above comment is correct, this should be malicious non recoverable

Reply:
Depends on the block type. For RPC blocks we check all signatures at once so the validity of the signature set is not deterministic on the block root